iceoryx_posh/test/integrationtests/test_service_discovery.cpp

```diff
@@ -17,58 +17,70 @@
+#include "iceoryx_posh/popo/listener.hpp"
 #include "iceoryx_posh/popo/untyped_publisher.hpp"
+#include "iceoryx_posh/popo/wait_set.hpp"
 #include "iceoryx_posh/runtime/posh_runtime.hpp"
 #include "iceoryx_posh/runtime/service_discovery.hpp"
 #include "iceoryx_posh/testing/roudi_gtest.hpp"
 #include "mocks/posh_runtime_mock.hpp"
 #include "test.hpp"
 
-
 #include <type_traits>
 
 namespace
 {
 using namespace ::testing;
 using namespace iox::runtime;
 using namespace iox::cxx;
+using namespace iox::popo;
+using namespace iox::capro;
 using iox::capro::IdString_t;
 using iox::capro::ServiceDescription;
 using iox::roudi::RouDiEnvironment;
 using iox::runtime::ServiceContainer;
 
+
 class ServiceDiscovery_test : public RouDi_GTest
 {
   public:
     iox::runtime::PoshRuntime* runtime{&iox::runtime::PoshRuntime::initRuntime("Runtime")};
     ServiceDiscovery sut;
-};
+    static bool callbackWasCalled;
+    static ServiceContainer serviceContainer;
 
-TIMING_TEST_F(ServiceDiscovery_test, GetServiceRegistryChangeCounterOfferStopOfferService, Repeat(5), [&] {
-    auto serviceCounter = sut.getServiceRegistryChangeCounter();
-    auto initialCout = serviceCounter->load();
-
-    iox::popo::UntypedPublisher pub({"service", "instance", "event"});
+    void SetUp()
+    {
+        callbackWasCalled = false;
+        serviceContainer.clear();
+    }
 
-    TIMING_TEST_EXPECT_TRUE(initialCout + 1 == serviceCounter->load());
+    static void testCallback(ServiceDiscovery* const serviceDiscoveryPointer)
+    {
+        IOX_DISCARD_RESULT(serviceDiscoveryPointer);
+        callbackWasCalled = true;
+    }
 
-    pub.stopOffer();
-    this->InterOpWait();
+    static void searchForService(ServiceDiscovery* const serviceDiscoveryPointer, ServiceDescription* service)
+    {
+        serviceContainer = serviceDiscoveryPointer->findService(
+            service->getServiceIDString(), service->getInstanceIDString(), service->getEventIDString());
+    }
+};
 
-    TIMING_TEST_EXPECT_TRUE(initialCout + 2 == serviceCounter->load());
-});
+bool ServiceDiscovery_test::callbackWasCalled{false};
+ServiceContainer ServiceDiscovery_test::serviceContainer;
 
-TEST_F(ServiceDiscovery_test, FindServiceWithWildcardsReturnsOnlyIntrospectionServices)
+TEST_F(ServiceDiscovery_test, FindServiceWithWildcardsReturnsOnlyIntrospectionServicesAndServiceRegistry)
 {
     ::testing::Test::RecordProperty("TEST_ID", "d944f32c-edef-44f5-a6eb-c19ee73c98eb");
-    auto serviceContainer = sut.findService(iox::capro::Wildcard, iox::capro::Wildcard, iox::capro::Wildcard);
-    ASSERT_FALSE(serviceContainer.has_error());
+    auto searchResult = sut.findService(iox::capro::Wildcard, iox::capro::Wildcard, iox::capro::Wildcard);
 
-    auto searchResult = serviceContainer.value();
 
     for (auto& service : searchResult)
     {
-        EXPECT_THAT(service.getServiceIDString().c_str(), StrEq("Introspection"));
+        EXPECT_THAT(service.getInstanceIDString().c_str(), StrEq("RouDi_ID"));
```

Comment:
The sudden capitalization in `RouDi` is not reader-friendly. Not sure we should get rid of this by using a lower case `d`.